### ISSUE TITLE
[BUG] Unable to patch image when failed to communicate with buildkitd

### ISSUE THREAD

**Issue by ritazh** [project-copacetic/copacetic#157]
### Version of copa

v0.2.0

### Expected Behavior

Patch succeeds

### Actual Behavior

```console
Starting image patching
Error: failed to dial gRPC: rpc error: code = Unavailable desc = rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /run/buildkit/buildkitd.sock: connect: no such file or directory"
Finished image patching
time="2023-05-21T22:35:21Z" level=info msg="auto snapshotter: using overlayfs"
time="2023-05-21T22:35:21Z" level=warning msg="using host network as the default"
time="2023-05-21T22:35:21Z" level=info msg="found worker \"q5k6257l21nsa2apkvri5g60b\", labels=map[org.mobyproject.buildkit.worker.executor:oci org.mobyproject.buildkit.worker.hostname:vm1 org.mobyproject.buildkit.worker.network:host org.mobyproject.buildkit.worker.oci.process-mode:sandbox org.mobyproject.buildkit.worker.selinux.enabled:false org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/amd64/v2 linux/amd64/v3 linux/amd64/v4 linux/386]"
time="2023-05-21T22:35:21Z" level=warning msg="using host network as the default"
time="2023-05-21T22:35:21Z" level=info msg="found worker \"w4v4jlc5lrw5c8i6r42f4j909\", labels=map[org.mobyproject.buildkit.worker.containerd.namespace:buildkit org.mobyproject.buildkit.worker.containerd.uuid:91fde356-0f86-4d04-9899-109cbdb56f59 org.mobyproject.buildkit.worker.executor:containerd org.mobyproject.buildkit.worker.hostname:vm1 org.mobyproject.buildkit.worker.network:host org.mobyproject.buildkit.worker.selinux.enabled:false org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/amd64/v2 linux/amd64/v3 linux/amd64/v4 linux/386]"
time="2023-05-21T22:35:21Z" level=info msg="found 2 workers, default=\"q5k6257l21nsa2apkvri5g60b\""
time="2023-05-21T22:35:21Z" level=warning msg="currently, only the default worker can be used."
time="2023-05-21T22:35:21Z" level=info msg="stopping server"
buildkitd: context canceled
```

### Steps To Reproduce

Followed the steps from https://project-copacetic.github.io/copacetic/website/quick-start#sample-steps
This seems to be a sporadic occurrence when there's an issue with buildkit. 
Ideally copa should be fault tolerant.

**Comment by smurawski** [comment]
Digging further, there seems to be a provision for retries, but the client is initialized in fail fast.  It's cool if that's the expected behavior, but having an option for retries or at least a more specific error message (rather than just returning the buildkit client error) would be useful for newer users.


https://github.com/project-copacetic/copacetic/blob/main/pkg/buildkit/buildkit.go#L131
```
config.Client, err = client.New(ctx, buildkitAddr, client.WithFailFast())
	if err != nil {
		return nil, err
	}
````

**Comment by sozercan** [comment]
fixed in #239
